--- a/mysql-mmm.spec	2018-05-04 18:55:29.000000000 +0000
+++ b/mysql-mmm.spec	2018-11-17 14:31:19.982837485 +0000
@@ -24,6 +24,9 @@
 Patch0: mysql-mmm-2.1.0-paths.patch
 Patch1: mysql-mmm-fix-bug-with-newer-net-arp.patch
 Patch2: mysql-mmm-fix-cve-remote-command-injection.patch
+Patch3: mmm_aws.patch
+Patch4: mmm_bin_aws.patch
+Patch5: mmm_multi_source.patch
 
 %description
 MMM (MySQL Master-Master Replication Manager) is a set of flexible scripts
@@ -85,6 +88,9 @@
 %patch0 -p1
 %patch1 -p1
 %patch2 -p1
+%patch3 -p1
+%patch4 -p1
+%patch5 -p1
 
 
 %build
