--- mysql-mmm-2.2.1/bin/agent/check_ip	2010-05-07 03:55:50.000000000 -0400
+++ mysql-mmm-2.2.2/bin/agent/check_ip	2017-11-16 04:31:43.932026504 -0500
@@ -6,18 +6,19 @@
 use MMM::Agent::Helpers::Actions;
 
 # Check arguments
-if (scalar(@ARGV) != 3) {
-	print "Usage: $0 <config_file> <interface> <ip>\n\n";
+if (scalar(@ARGV) != 4) {
+	print "Usage: $0 <config_file> <interface> <eni> <ip>\n\n";
 	exit(1);
 }
 
 # Fetch arguments
 my $config_file = shift;
 my $if = shift;
+my $eni = shift;
 my $ip = shift;
 
 # Finally do the work
-MMM::Agent::Helpers::Actions::check_ip($if, $ip);
+MMM::Agent::Helpers::Actions::check_ip($if, $eni, $ip);
 
 __END__
 
--- mysql-mmm-2.2.1/bin/agent/clear_ip	2010-05-07 03:55:50.000000000 -0400
+++ mysql-mmm-2.2.2/bin/agent/clear_ip	2017-11-16 04:32:19.554026020 -0500
@@ -6,18 +6,19 @@
 use MMM::Agent::Helpers::Actions;
 
 # Check arguments
-if (scalar(@ARGV) != 3) {
-	print "Usage: $0 <config_file> <interface> <ip>\n\n";
+if (scalar(@ARGV) != 4) {
+	print "Usage: $0 <config_file> <interface> <eni> <ip>\n\n";
 	exit(1);
 }
 
 # Fetch arguments
 my $config_file = shift;
 my $if = shift;
+my $eni = shift;
 my $ip = shift;
 
 # Finally do the work
-MMM::Agent::Helpers::Actions::clear_ip($if, $ip);
+MMM::Agent::Helpers::Actions::clear_ip($if, $eni, $ip);
 
 __END__
 
--- mysql-mmm-2.2.1/bin/agent/configure_ip	2010-05-07 03:55:50.000000000 -0400
+++ mysql-mmm-2.2.2/bin/agent/configure_ip	2017-11-16 04:32:49.035025619 -0500
@@ -6,18 +6,19 @@
 use MMM::Agent::Helpers::Actions;
 
 # Check arguments
-if (scalar(@ARGV) != 3) {
-	print "Usage: $0 <config_file> <interface> <ip>\n\n";
+if (scalar(@ARGV) != 4) {
+	print "Usage: $0 <config_file> <interface> <eni> <ip>\n\n";
 	exit(1);
 }
 
 # Fetch arguments
 my $config_file = shift;
 my $if = shift;
+my $eni = shift;
 my $ip = shift;
 
 # Finally do the work
-MMM::Agent::Helpers::Actions::configure_ip($if, $ip);
+MMM::Agent::Helpers::Actions::configure_ip($if, $eni, $ip);
 
 __END__
 
